jetson/nav/environment.cpp

```diff
@@ -12,10 +19,143 @@ Obstacle Environment::getObstacle() {
+    if (rover->autonState().is_auton) {
+        bool rightReady = isRightTargetFilterReady();
+        bool leftReady = isLeftTargetFilterReady();
+        if (rightReady){
+            std::cout << "right ready: " << mTargetRight.id << std::endl;
+            if (mTargetRight.id == baseGateID){
+                std::cout << "updated post 1" << std::endl;
+                mPostOne = createOdom(rover->odometry(), mRightBearingFilter.get(), mRightDistanceFilter.get(), rover);
+                hasPostOne = true;
+            }
+            if (mTargetRight.id == baseGateID + 1){
+                std::cout << "updated post 2" << std::endl;
+                mPostTwo = createOdom(rover->odometry(), mRightBearingFilter.get(), mRightDistanceFilter.get(), rover);
+                hasPostTwo = true;
+            }
+        }
+        if (leftReady){
+            std::cout << "left ready: " << mTargetLeft.id << std::endl;
+            std::cout << "course waypoint id: " << course->getCurrentWaypoint().id << std::endl;
+            if (mTargetLeft.id == baseGateID){
+                std::cout << "updated post 1" << std::endl;
+                mPostOne = createOdom(rover->odometry(), mLeftBearingFilter.get(), mLeftDistanceFilter.get(), rover);
+                hasPostOne = true;
+            }
+            if (mTargetLeft.id == baseGateID + 1){
+                std::cout << "updated post 2" << std::endl;
+                mPostTwo = createOdom(rover->odometry(), mLeftBearingFilter.get(), mLeftDistanceFilter.get(), rover);
+                hasPostTwo = true;
+            }
+        }
+    } else {
+        hasPostOne = hasPostTwo = false;
+        double cosine = cos(degreeToRadian(rover->odometry().latitude_deg, rover->odometry().latitude_min));
+        rover->setLongMeterInMinutes( 60 / (EARTH_CIRCUM * cosine / 360) );
+    }
+
+    
+}
+
+bool Environment::hasGateLocation() const {
+    return hasPostOne && hasPostTwo;
+}
+
+bool Environment::hasPostOneLocation() const {
+    return hasPostOne;
+}
+
+bool Environment::hasPostTwoLocation() const {
+    return hasPostTwo;
+}
+
+Odometry Environment::getPostOneLocation() {
+    return mPostOne;
+}
+
+Odometry Environment::getPostTwoLocation() {
+    return mPostTwo;
+}
+
+Vector2d Environment::getPostOneRelative(Odometry cur) {
+    double bearing = degreeToRadian(calcBearing(cur, mPostOne));
+    double distance = estimateNoneuclid(cur, mPostOne);
+    return {distance * cos(bearing), distance * sin(bearing)};
+}
+
+Vector2d Environment::getPostTwoRelative(Odometry cur) {
+    double bearing = degreeToRadian(calcBearing(cur, mPostTwo));
+    double distance = estimateNoneuclid(cur, mPostTwo);
+    return {distance * cos(bearing), distance * sin(bearing)};
+}
+
+bool Environment::areTargetFiltersReady() const {
+    return mLeftDistanceFilter.ready() && mRightDistanceFilter.ready() && mLeftBearingFilter.ready() && mRightBearingFilter.ready();
```

Comment:
Any reason why you aren't just calling `isLeftTargetFilterReady()` and `isRightTargetFilterReady()`?